src/main/java/org/intellij/sonar/configuration/partials/SonarResourcesTableView.java

```diff
@@ -19,53 +23,52 @@
 import com.intellij.util.ui.ColumnInfo;
 import com.intellij.util.ui.ListTableModel;
 import org.intellij.sonar.configuration.ResourcesSelectionConfigurable;
+import org.intellij.sonar.persistence.Resource;
 import org.jetbrains.annotations.Nullable;
-import org.sonar.wsclient.services.Resource;
 
 public class SonarResourcesTableView {
 
   private final TableView<Resource> mySonarResourcesTable;
   private final JComponent myJComponent;
   private final SonarServersView mySonarServersView;
   private final Project myProject;
-  private static final ColumnInfo<Resource,String> TYPE_COLUMN = new ColumnInfo<Resource,String>("Type") {
+  private static final Map<String, String> TYPE_BY_QUALIFIER = new HashMap<>();
```

Comment:
we can use ImmutableMap from guava here:
```
ImmutableMap.builder()
      .put(PROJECT.getQualifier(), "Project")
      .put(MODULE.getQualifier(), "Module")
      .build();
```

Reply:
Ah, nice, I'm not used to Guava, I think I should definitely have a look at it. Though I'll use the typesafe variant `ImmutableMap.<String, String>builder()`. :-)